### ISSUE TITLE
Warnings from PyCharm

### ISSUE THREAD

**Issue by ThGravo** [ashleve/lightning-hydra-template#173]
PyCharm's code analysis tool give me a couple of warning:
```
tests/unit/test_mnist_datamodule.py
    Warning:Warning:line (23)Expected type 'Sized', got 'Optional[Dataset]' instead
    Warning:Warning:line (23)Expected type 'Sized', got 'Optional[Dataset]' instead
    Warning:Warning:line (23)Expected type 'Sized', got 'Optional[Dataset]' instead
src/utils/utils.py
    Warning:Warning:line (120)Cannot find reference 'loggers' in '__init__.py'
    Warning:Warning:line (149)Expected type 'Namespace', got 'dict[str, Union[int, Any]]' instead
    Warning:Warning:line (163)Cannot find reference 'loggers' in '__init__.py'
    Warning:Warning:line (169)Cannot find reference 'loggers' in '__init__.py'
src/callbacks/wandb_callbacks.py
    Warning:Warning:line (264)Unresolved attribute reference 'datamodule' for class 'Trainer'
    Warning:Warning:line (270)Unexpected argument
```
Is that something I should be concerned about?

**Comment by ashleve** [comment]
Hi, 

some variables in pytorch lightning are assigned on the run without any static typing, which is why pycharm sees them as a possible mismatch. In other examples received type is different than expected but it doesn't matter since its interface is the same.

I don't think there's anything to worry about here

**Comment by ThGravo** [comment]
Also not with the last one: `Warning:Warning:line (270)Unexpected argument` ?
Corresponding code is: `preds = torch.argmax(logits, axis=-1)`

**Comment by ashleve** [comment]
@ThGravo Thanks, `axis` is not an expected argument but it still somehow works correctly. Perhaps it's an old interface.
I will change it to `torch.argmax(logits, dim=-1)` since it gives the same result

**Comment by ThGravo** [comment]
@ashleve Thanks!
